src/dc_receive_imf.rs

```diff
@@ -32,17 +32,29 @@ enum CreateEvent {
     IncomingMsg,
 }
 
+pub async fn dc_receive_imf(
+    context: &Context,
+    imf_raw: &[u8],
+    server_folder: impl AsRef<str>,
+    server_uid: u32,
+    seen: bool,
+) -> Result<()> {
+    dc_receive_imf_inner(context, imf_raw, server_folder, server_uid, seen, false).await
+}
+
 /// Receive a message and add it to the database.
 ///
 /// Returns an error on recoverable errors, e.g. database errors. In this case,
 /// message parsing should be retried later. If message itself is wrong, logs
 /// the error and returns success.
-pub async fn dc_receive_imf(
+pub async fn dc_receive_imf_inner(
+    // TODO this is not a nice solution. dc_receive_imf_inner is not a nice name, either, dc_receive_imf_fetching_existing would be very long.
     context: &Context,
     imf_raw: &[u8],
     server_folder: impl AsRef<str>,
     server_uid: u32,
     seen: bool,
+    fetching_existing_messages: bool,
 ) -> Result<()> {
```

Comment:
i think passing the flag down to all the class is fine here.

wrt dc_receive_imf_inner() - maybe, instead of adding this function, just add an additional parameter directly to dc_receive_imf()?

Reply:
I tried and got 100 errors from all the tests using dc_receive_imf() and did not want to change them all. What we could do is create an fn `dc_receive_inner_tst()` used in tests without fetching_existing_messages param. This would be a big change too but it could be done by the rename feature of my IDE so no work.

Reply:
let's see what to do about it at a later point, it is not bad currently. and it's only a small aspect and probably produces too much noise.